src/widgets/editor.rs

```diff
@@ -118,9 +120,77 @@ impl Future for Editor {
 
 impl Editor {
     /// Handle keyboard and mouse events
-    pub fn handle_input(&mut self, event: TermionEvent) {
-        if let Some(view) = self.views.get_mut(&self.current_view) {
-            view.handle_input(event)
+    pub fn handle_input(&mut self, event: Event) {
+        // We have to remove and insert again, to beat the borrow-checker
+        match event {
+            Event::Key(key) => {
+                match self.keybindings.keymap.get(&event).cloned() {
+                    Some(cmd) => self.handle_command(cmd),
+                    None => { 
+                        if let Some(view) = self.views.get_mut(&self.current_view) {
+                            match key {
```

Comment:
This logic could be moved to the view itself.

Reply:
It is on HEAD